python/tidy/servo_tidy_tests/test_tidy.py

```diff
@@ -181,6 +181,17 @@ def test_modeline(self):
         self.assertEqual('emacs file variables present', errors.next()[2])
         self.assertNoMoreErrors(errors)
 
+    def test_html_with_empty_title_tag(self):
+        errors = tidy.collect_errors_for_files(iterFile('empty_title_tag.html'), [tidy.check_html_title_tag], [], print_text=False)
+        error = errors.next()
+        self.assertEqual(error[1], 2)
+        self.assertEqual(error[2], 'Found empty title tag. Please add descriptive title.')
+        self.assertNoMoreErrors(errors)
+
+    def test_html_with_no_title_tag(self):
+        errors = tidy.collect_errors_for_files(iterFile('no_title_tag.html'), [tidy.check_html_title_tag], [], print_text=False)
```

Comment:
Note that the self-test fails here because you're feeding the html-checker to `checking_functions` instead of `line_checking_functions`